### PR TITLE
[optimize] Acm enhance base Spring Cloud Common 2.2.1

### PR DESCRIPTION
**Pull request by yuhuangbin**
### Describe what this PR does / why we need it

- optimize Acm module base Spring Cloud Common 2.2.1
- solved `NacosPropertySourceRepository` data is not updated when data changes are monitored

### CODE REVIEW

File: spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/AcmPropertySourceRepository.java [hunk 1]
```diff
@@ -17,54 +17,43 @@
...
 
 /**
  * @author juven.xuxb, 5/17/16.
  */
 public class AcmPropertySourceRepository {
 
-	private final ApplicationContext applicationContext;
+	private static final Set<AcmPropertySource> ACM_PROPERTY_SOURCE_REPOSITORY = new CopyOnWriteArraySet();
+
+	private AcmPropertySourceRepository () {
```
Comment (echooymxq) [inline spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/AcmPropertySourceRepository.java]: The `PropertySourceRepository` should not be used like a tool class. Violated the original intention of `Repository`.

File: spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/endpoint/AcmEndpoint.java [hunk 2]
```diff
@@ -43,20 +44,16 @@
...
 	private ThreadLocal<DateFormat> dateFormat = new ThreadLocal<DateFormat>() {
 		@Override
 		protected DateFormat initialValue() {
 			return new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
 		}
 	};
 
-	public AcmEndpoint(AcmProperties properties, AcmRefreshHistory refreshHistory,
-			AcmPropertySourceRepository propertySourceRepository) {
+	public AcmEndpoint(AcmProperties properties, AcmRefreshHistory refreshHistory) {
```
Comment (echooymxq) [inline spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/endpoint/AcmEndpoint.java]: It better not remove it.

File: spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/endpoint/AcmEndpointAutoConfiguration.java [hunk 3]
```diff
@@ -43,22 +42,17 @@
...
-	@Autowired
-	private AcmPropertySourceRepository acmPropertySourceRepository;
-
 	@ConditionalOnMissingBean
 	@ConditionalOnEnabledEndpoint
 	@Bean
 	public AcmEndpoint acmEndpoint() {
-		return new AcmEndpoint(acmProperties, acmRefreshHistory,
-				acmPropertySourceRepository);
+		return new AcmEndpoint(acmProperties, acmRefreshHistory);
```
Comment (echooymxq) [inline spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/endpoint/AcmEndpointAutoConfiguration.java]: The same as above.

File: spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/endpoint/AcmEndpointAutoConfiguration.java [hunk 4]
```diff
@@ -43,22 +42,17 @@
...
 	}
 
 	@Bean
 	@ConditionalOnMissingBean
-	public AcmHealthIndicator acmHealthIndicator(AcmProperties acmProperties,
-			AcmPropertySourceRepository acmPropertySourceRepository) {
-		return new AcmHealthIndicator(acmProperties, acmPropertySourceRepository);
+	public AcmHealthIndicator acmHealthIndicator(AcmProperties acmProperties) {
```
Comment (echooymxq) [inline spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/endpoint/AcmEndpointAutoConfiguration.java]: The same as above.

File: spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/endpoint/AcmHealthIndicator.java [hunk 5]
```diff
@@ -36,18 +37,13 @@
 
 	private final AcmProperties acmProperties;
 
-	private final AcmPropertySourceRepository acmPropertySourceRepository;
-
 	private final List<String> dataIds;
 
-	public AcmHealthIndicator(AcmProperties acmProperties,
-			AcmPropertySourceRepository acmPropertySourceRepository) {
+	public AcmHealthIndicator(AcmProperties acmProperties) {
```
Comment (echooymxq) [inline spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/endpoint/AcmHealthIndicator.java]: The same as above.

**Comment by echooymxq** [comment]
@yuhuangbin  And I think that it can add a `AcmConfigBootstrapConfiguration` to replace `AcmPropertySourceLocator` configure in the spring.factories files.

File: spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/AcmPropertySourceRepository.java (hunk 1 above)
Comment (fangjian0423) [reply spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/AcmPropertySourceRepository.java]: yes, the name is really not very reasonable, as `NacosPropertySourceRepository` do.

Comment (echooymxq) [reply spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/AcmPropertySourceRepository.java]: `DubboServiceMetadataRepository` is a better example.

File: spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/refresh/AcmContextRefresher.java [hunk 6]
```diff
@@ -82,14 +82,15 @@ private void registerDiamondListenersForApplications() {
...
+				registerDiamondListener(dataId,
+						acmIntegrationProperties.getAcmProperties().getGroup());
 			}
 		}
 	}
 
-	private void registerDiamondListener(final String dataId) {
-
-		ConfigChangeListener listener = listenerMap.computeIfAbsent(dataId,
+	private void registerDiamondListener(final String dataId, String group) {
```
Comment (echooymxq) [inline spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/refresh/AcmContextRefresher.java]: The `group` arg should final.

File: spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/endpoint/AcmHealthIndicator.java [hunk 7]
```diff
@@ -36,18 +36,18 @@
 
 	private final AcmProperties acmProperties;
 
-	private final AcmPropertySourceRepository acmPropertySourceRepository;
-
 	private final List<String> dataIds;
 
+	private AcmPropertySourceRepository acmPropertySourceRepository;
```
Comment (echooymxq) [inline spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/endpoint/AcmHealthIndicator.java]: The `acmPropertySourceRepository`variable should final.

File: spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/endpoint/AcmEndpoint.java [hunk 8]
```diff
@@ -43,7 +43,7 @@
 
 	private final AcmRefreshHistory refreshHistory;
 
-	private final AcmPropertySourceRepository propertySourceRepository;
+	private AcmPropertySourceRepository acmPropertySourceRepository;
```
Comment (echooymxq) [inline spring-cloud-alicloud-acm/src/main/java/com/alibaba/alicloud/acm/endpoint/AcmEndpoint.java]: The acmPropertySourceRepositoryvariable should final.

File: spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/NacosPropertySourceRepository.java [hunk 9]
```diff
@@ -48,7 +48,7 @@ private NacosPropertySourceRepository() {
 	@Deprecated
 	public static void collectNacosPropertySources(
 			NacosPropertySource nacosPropertySource) {
-		NACOS_PROPERTY_SOURCE_REPOSITORY.putIfAbsent(nacosPropertySource.getDataId(),
+		NACOS_PROPERTY_SOURCE_REPOSITORY.put(nacosPropertySource.getDataId(),
```
Comment (echooymxq) [inline spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/NacosPropertySourceRepository.java]: Why modify it here? if Nacos Config has any problems, pull another request.

File: spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/NacosPropertySourceRepository.java [hunk 10]
```diff
@@ -65,9 +65,8 @@ public static NacosPropertySource getNacosPropertySource(String dataId) {
 
 	public static void collectNacosPropertySource(
 			NacosPropertySource nacosPropertySource) {
-		NACOS_PROPERTY_SOURCE_REPOSITORY
-				.putIfAbsent(getMapKey(nacosPropertySource.getDataId(),
-						nacosPropertySource.getGroup()), nacosPropertySource);
+		NACOS_PROPERTY_SOURCE_REPOSITORY.put(getMapKey(nacosPropertySource.getDataId(),
```
Comment (echooymxq) [inline spring-cloud-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/NacosPropertySourceRepository.java]: The same above.

**Pull request merged**